http.go

```diff
@@ -108,34 +108,37 @@ func (g *Gateway) GraphQLHandler(w http.ResponseWriter, r *http.Request) {
 		operations = append(operations, operation)
 		// or we got a POST request
 	} else if r.Method == http.MethodPost {
-		// read the full request body
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			payloadErr = fmt.Errorf("encountered error reading body: %s", err.Error())
-		}
+		body, fileMap, extractError := extractBody(r)
+		if extractError != nil {
+			payloadErr = extractError
+		} else {
```

Comment:
What do you think about moving this logic into `extractBody`? My thought is that `extractBody` can be responsible for pulling out the individual parts regardless of method.

Reply:
Thanks for having a look. Which part do you mean? Do you mean extractBody should take care of things done in  if/then/else or together with parsing requests with GET method? It seems logical to me to move all request parsing in a separate function(s) but I tried to do as little as possible to already existing code. If you don't mind, I'd refactor it in that way

Reply:
yep,  that's exactly what i had in mind! Let me go through this one more time but I think this is probably good to go.